network-api/networkapi/wagtailpages/pagemodels/customblocks/common/base_link_block.py

```diff
@@ -0,0 +1,135 @@
+
+    @property
+    def url(self):
+        link_to = self.get("link_to")
+
+        method = getattr(self, f"get_{link_to}_link")
+        return method()
+
+    def get_link_to(self):
+        """
+        Return link type for accessing in templates
+        """
+        return self.get("link_to")
+
+
+class BaseLinkBlock(blocks.StructBlock):
+    link_to = blocks.ChoiceBlock(
+        choices=[
+            ("page", "Page"),
+            ("external_url", "External URL"),
+            ("relative_url", "Relative URL"),
+            ("email", "Email"),
+            ("anchor", "Anchor"),
+        ],
+        required=False,
+        label="Link to",
+    )
+    page = blocks.PageChooserBlock(required=False, label="Page")
+    external_url = blocks.URLBlock(
+        max_length=300,
+        required=False,
+        label="External URL",
+        help_text="Enter a full URL including http:// or https://",
+    )
+    relative_url = blocks.CharBlock(
+        max_length=300,
+        required=False,
+        validators=[RelativeURLValidator()],
+        label="Relative URL",
+        help_text='A path relative to this domain. For example, "/foo/bar"',
+    )
+    anchor = blocks.CharBlock(
+        max_length=300,
+        required=False,
+        validators=[AnchorLinkValidator()],
+        label="#",
+        help_text='An id attribute of an element on the current page. For example, "#section-1"',
+    )
+    email = blocks.EmailBlock(required=False)
+
+    class Meta:
+        abstract = True
+        value_class = BaseLinkValue
+
+    def get_default_values(self):
+        return {
+            "page": None,
+            "external_url": "",
+            "relative_url": "",
+            "anchor": "",
+            "email": "",
+        }
+
+    def clean(self, value):
+        clean_values = super().clean(value)
+        errors = {}
+
+        url_default_values = self.get_default_values()
+        url_type = clean_values.get("link_to")
+
+        # Check that a value has been uploaded for the chosen link type
+        if url_type != "" and clean_values.get(url_type) in [None, ""]:
+            errors[url_type] = ErrorList(["You need to add a {} link".format(url_type.replace("_", " "))])
+        else:
+            try:
+                # Remove values added for link types not selected
+                url_default_values.pop(url_type, None)
+                for field in url_default_values:
+                    clean_values[field] = url_default_values[field]
+            except KeyError:
+                errors[url_type] = ErrorList(["Enter a valid link type"])
+
+        if errors:
+            raise blocks.StreamBlockValidationError(block_errors=errors, non_block_errors=ErrorList([]))
+
+        return clean_values
+
+
+class BaseLinkBlockAdapter(StructBlockAdapter):
+    """Custom adapter to register JS for conditionally hiding/showing fields.
+
+    Can be used with any block that extends `BaseAbstractLinkBlock`.
```

Comment:
Should this comment be for extending `BaseLinkBlock`?

Reply:
Thanks Rob! That's fixed now